src/lxml/sax.py

```diff
@@ -165,6 +165,7 @@ def __init__(self, element_or_tree, content_handler):
         from xml.sax.xmlreader import AttributesNSImpl as attr_class
         self._attr_class = attr_class
         self._empty_attributes = attr_class({}, {})
+        self._prefix_map = dict((v, k) for (k, v) in element.nsmap.items())
```

Comment:
I think it should check the prefix map of the element that it currently processes, not the root element.

Reply:
Sounds reasonable, I'll look into that.